### PR TITLE
Customize paths

### PR DESCRIPTION
**Pull request by craig-mulligan**
@brownjohnf, So figured out I do need `apps` to register custom helpers. So I've updated our ghost-fork so that all paths are configurable. 

So what'd we do is set the `GHOST_CONTENT_PATH` and then explicitly set `GHOST_THEME_PATH` and `GHOST_APPS_PATH`.

I've also had to upgrade to node 6 because `ghost-apps` doesn't work in node 4.* and I didn't feel like forking another ghost module. 

Another thing we'll have to do is update the `activeApps` + `installedApp` key in the `Settings` table in the db from `resin-mixpanel` to `resin-helpers` (they don't have a way to do it via ui.

Have I missed something? Hopefully this is the last of it and deploying is a breeze from now on ✈️ 🌬

### CODE REVIEW

File: .dockerenv [hunk 1]
```diff
@@ -1,7 +1,10 @@
+NODE_ENV=development
 GHOST_URL=http://localhost:2368/blog
 GHOST_PORT=2368
-GHOST_SQLITE_PATH=content/data/ghost.db
-GHOST_DATA_PATH=content
+GHOST_SQLITE_PATH=/data/content/ghost.db
```
Comment (brownjohnf) [inline .dockerenv]: This should stay content/data/ghost.db. It would be invalid to share the sqlite database. If using sqlite, you can't put it on /data.

Comment (brownjohnf) [inline .dockerenv]: Even better, for mimicking production, would be to add a docker-compose file that actually spins mysql for ghost to use.

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,4 @@ npm-debug.log
 .DS_Store
 *.swp
 *.swo
+run.sh
```
Comment (brownjohnf) [inline .gitignore]: What is run.sh? Anything that you're using locally, I'd suggest namespacing. Someone may want to actually use a run.sh some day as part of the repo.

File: Makefile [hunk 3]
```diff
@@ -3,16 +3,28 @@ IMAGE=localhost/resin-blog:latest
 build:
 	docker build -t ${IMAGE} .
 
+# runs an image with bind-mounts for local work
 run: build
 	docker run -it --rm -p 2368:2368 \
+		-v $$(pwd)/content:/data/content \
```
Comment (brownjohnf) [inline Makefile]: I thought you wanted to revert this?

File: config.js [hunk 4]
```diff
@@ -29,9 +29,13 @@ if (process.env.GHOST_DB_ENGINE === "mysql") {
 }
 
 ghostConfig = {
-        url: process.env.GHOST_URL,
+    url: process.env.GHOST_URL,
     paths: {
-        contentPath: process.env.GHOST_DATA_PATH
+        contentPath: process.env.GHOST_CONTENT_PATH,
+        dataPath: process.env.GHOST_DATA_PATH,
```
Comment (brownjohnf) [inline config.js]: It seems like these should default to GHOST_CONTENT_PATH/<dir> unless overridden.

Comment (craig-mulligan) [inline config.js]: They do, but logic is in resin-ghost: https://github.com/resin-io-modules/Ghost/blob/stable/core/server/config/index.js#L176

**Comment by craig-mulligan** [comment]
ping @brownjohnf

File: config.js (hunk 4 above)
Comment (craig-mulligan) [inline config.js]: I take that back, that check actually doesn't work. Because they first feed the config generator some default paths fisrt this._config.paths.themePath is never undefined. 

So agreed we should defaults here.

**Comment by brownjohnf** [comment]
@craig-mulligan should this still be merged? I'm in the process of getting the blog deployed on k8s, but at this point i'm confused about which version I should be working with.

**Comment by craig-mulligan** [comment]
@brownjohnf it depends if we need custom for themes. Remember we had to extend ghost, because it only allowed you to configure the path of `/content` and not it's sub directories eg `/themes`. If that is not longer a constraint then we can use standard `ghost` and not our forked version. 

The other thing this PR adds is the resin helper app, which handles provides tokens to the front end analytics. I can't remember why it's included here or why it's needed. 

Basically if you tell me where we need the custom paths stuff in k8s then I can update this PR if we do need it then we should use this branch.

**Comment by craig-mulligan** [comment]
ping @brownjohnf

**Comment by brownjohnf** [comment]
@craig-mulligan I think we do still need separate paths, because we want the theme data to be in the container, not persisted. Right?

The thing confusing me is that it looks like you pushed a bunch of changes to the resin-on-resin staging blog, but they're not reflected in this PR. I'd just like to know which HEAD I should be working from.

**Comment by craig-mulligan** [comment]
Hey @brownjohnf, 

So this is the version we should be going with. I've move it back to ghost instead of our forked version. One thing it does add is a few custom handlebar helpers that we need. 

In order to add these helpers we had to add an app which requires a db change. To do it we have to change the activeApps key in the Settings table in the db from `resin-mixpanel` to `resin-helpers`.

If you like you can just send me the db you are working with and I'll update it.

**Comment by brownjohnf** [comment]
Is there no support for anything like migrations? We're running against a mysql db in AWS RDS

**Comment by craig-mulligan** [comment]
Not by the sounds of it, their instructions are here: https://github.com/TryGhost/Ghost/wiki/Apps-Getting-Started-for-Ghost-Devs#installing

We could rename the app `resin-mixpanel` because that's already in the db, but it's not semantically correct ofc.

**Comment by craig-mulligan** [comment]
@brownjohnf does the above make sense? Ping me if you need anything.

**Comment by brownjohnf** [comment]
Makes sense, but is just ridiculous. I'm very unimpressed with ghost.

**Comment by brownjohnf** [comment]
This needs to be rebased to pass tests. Just FYI for future readers.
